python/cudf/cudf_pandas_tests/test_cudf_pandas.py

```diff
@@ -1421,3 +1422,61 @@ def test_holidays_within_dates(holiday, start, expected):
             utc.localize(xpd.Timestamp(start)),
         )
     ) == [utc.localize(dt) for dt in expected]
+
+
+def test_cudf_pandas_debugging_different_results(monkeypatch):
+    cudf_mean = cudf.Series.mean
+
+    def mock_mean_one(self, *args, **kwargs):
+        return np.float64(1.0)
+
+    with monkeypatch.context() as monkeycontext:
+        monkeypatch.setattr(xpd.Series.mean, "_fsproxy_fast", mock_mean_one)
+        monkeycontext.setenv("CUDF_PANDAS_DEBUGGING", "True")
+        s = xpd.Series([1, 2])
+        with pytest.warns(
+            UserWarning,
+            match="The results from cudf and pandas were different.",
+        ):
+            assert s.mean() == 1.0
+    monkeypatch.setattr(xpd.Series.mean, "_fsproxy_fast", cudf_mean)
```

Comment:
At the end of each test, I'm patching `_fsproxy_fast` to explicitly undo the first patch made in the test

Reply:
Can you add a comment explaining this? Otherwise my future self (and other readers) will be confused why this is needed in addition to the monkeypatch context. Something like this, on each of the tests:
```suggestion
    # Must explicitly undo the patch. Proxy dispatch doesn't work with monkeypatch contexts.
    monkeypatch.setattr(xpd.Series.mean, "_fsproxy_fast", cudf_mean)
```